app/controllers/components/course/controller_component_host.rb

```diff
@@ -52,6 +53,25 @@ def gamified?
       def can_be_disabled?
         true
       end
+
+      # Returns a model which the current component can use to interface with its persisted
+      # settings.
+      #
+      # Example:
+      # If the component Course::FoobarComponent has settings, define a class
```

Comment:
@kxmbrian I think you can document the conventions at the top, but can leave the `Returns a model ... here`. Ideally we should make it easy for new developers to just read the top set of documentation and be able to create a component.